entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -78,38 +84,39 @@ class EntityHandler(
             .map {
                 compactEntities(it)
             }
-            .flatMap {
-                ok().body(BodyInserters.fromValue(serializeObject(it)))
+            .map {
+                ResponseEntity.ok().body(serializeObject(it))
             }
     }
 
     /**
      * Implements 6.5.3.1 - Retrieve Entity
      */
-    fun getByURI(req: ServerRequest): Mono<ServerResponse> {
-        val uri = req.pathVariable("entityId")
-        return uri.toMono()
+    @GetMapping("/{entityId}")
+    fun getByURI(@PathVariable entityId: String): Mono<ResponseEntity<String>> {
+        return entityId.toMono()
             .map {
-                if (!entityService.exists(uri)) throw ResourceNotFoundException("Entity Not Found")
+                if (!entityService.exists(entityId)) throw ResourceNotFoundException("Entity Not Found")
                 entityService.getFullEntityById(it)
             }
             .map {
                 it.compact()
             }
-            .flatMap {
-                ok().body(BodyInserters.fromValue(serializeObject(it)))
+            .map {
+                ResponseEntity.ok().body(serializeObject(it))
             }
     }
 
     /**
      * Implements 6.6.3.1 - Append Entity Attributes
      *
      */
-    fun appendEntityAttributes(req: ServerRequest): Mono<ServerResponse> {
-        val entityId = req.pathVariable("entityId")
-        val disallowOverwrite = req.queryParam("options").map { it == "noOverwrite" }.orElse(false)
-        val contextLink = extractContextFromLinkHeader(req)
-        return req.bodyToMono<String>()
+    @PostMapping("/{entityId}/attrs")
+    fun appendEntityAttributes(@RequestHeader httpHeaders: HttpHeaders, @PathVariable entityId: String, @RequestParam options: Optional<String>, @RequestBody attributes: Mono<String>): Mono<ResponseEntity<String>> {

```

Comment:
use the same name everywhere for the `@RequestBody` parameter (it is `entity` in `create` and `attributes` here and below). I would simply call it `body`.